### PR TITLE
Sort docs by `group`, `group_index`

### LINKED ISSUE #143: Group by including unrelated api routes
routes displayed inside unrelated routes section.

groupby

### PR DESCRIPTION
**Pull request by kitloong**
Fix #143

### CODE REVIEW

File: tests/LRDTest.php [hunk 1]
```diff
@@ -70,180 +70,240 @@ public function testGroupByURI()
...
+                    'group'       => 'welcome',
+                    'group_index' => 3
                 ]
-            ]
+            ],
         ];
         $this->assertSame($expected, $grouped->toArray());
     }
 
+    public function testGroupByURISorted()
```
Comment (kitloong) [inline tests/LRDTest.php:191]: @kevincobain2000 Please check the expected output from this PR

File: src/LaravelRequestDocs.php [hunk 2]
```diff
@@ -119,15 +119,23 @@ public function filterByMethods($docs, $get, $post, $put, $path, $delete, $head)
 
     public function groupDocs($docs, $group = 'default')
     {
+        if ($group === 'default') {
```
Comment (kevincobain2000) [inline src/LaravelRequestDocs.php:122]: better than before, but still not quite there.
Screen Shot 2023-02-25 at 17 28 30


```
    {
        "uri": "api/v1/health",
        "methods": [
            "GET"
        ],
        "middlewares": [
            "api"
        ],
....
        ],
        "group": "api/v1/health",
        "group_index": 0
    },
    {
        "uri": "api/v1/not-health",
        "methods": [
            "GET"
        ],
```



merged but will reopen the issue #143 
Won't push new tag yet.

Comment (kevincobain2000) [reply src/LaravelRequestDocs.php:122]: Loop on UI is here

https://github.com/rakutentech/laravel-request-docs/blob/374f4f16cf0d618c3905aa9f5750432e213bcd6b/ui/src/components/Sidebar.tsx#L22-L43

File: src/LaravelRequestDocs.php [hunk 3]
```diff
@@ -119,15 +119,23 @@ public function filterByMethods($docs, $get, $post, $put, $path, $delete, $head)
...
+            $groupDocs = $this->groupDocsByAPIURI($docs);
         }
 
         if ($group === 'controller_full_path') {
-            return $this->groupDocsByFQController($docs);
+            $groupDocs = $this->groupDocsByFQController($docs);
         }
 
-        return $docs;
+        return collect($groupDocs)->sortBy(['group', 'group_index'])
```
Comment (kevincobain2000) [inline src/LaravelRequestDocs.php:136]: Probably this

```

        return collect($groupDocs)->groupBy(['group', 'group_index'])
            ->values()
            ->flatten(2)
            ->toArray();
```

Comment (kevincobain2000) [reply src/LaravelRequestDocs.php:136]:
```
        $groupBy = ['group', 'group_index'];
        return collect($groupDocs)->groupBy($groupBy)
            ->values()
            ->flatten(count($groupBy))
            ->toArray();
```

Comment (kevincobain2000) [reply src/LaravelRequestDocs.php:136]: Fixes here #150

Comment (kevincobain2000) [reply src/LaravelRequestDocs.php:136]: ah UT failed, so will revert changes https://github.com/rakutentech/laravel-request-docs/pull/150/commits/c93806366822cfaee6f4b1d9f99d78d99056f84e 
Will wait for your pull req

**Pull request merged**